returns/methods/cond.py

```diff
@@ -1,22 +1,22 @@
 from typing import Type, TypeVar
 
-from returns.interfaces.specific.result import ResultLikeN
+from returns.interfaces.failable import DiverseFailableN
 from returns.primitives.hkt import Kind2, kinded
 
 _ValueType = TypeVar('_ValueType')
 _ErrorType = TypeVar('_ErrorType')
 
-_ResultKind = TypeVar('_ResultKind', bound=ResultLikeN)
+_DiverseFailableKind = TypeVar('_DiverseFailableKind', bound=DiverseFailableN)
 
 
 def internal_cond(
-    container_type: Type[_ResultKind],
+    container_type: Type[_DiverseFailableKind],
     is_success: bool,
     success_value: _ValueType,
     error_value: _ErrorType,
-) -> Kind2[_ResultKind, _ValueType, _ErrorType]:
+) -> Kind2[_DiverseFailableKind, _ValueType, _ErrorType]:
```

Comment:
Looks like this is not 100% related, but.

Let's use `KindN` here. Because I can see a bug in `Kind2` here. 
Here's a failing test:

```yml
- case: unify_reader_future_result
  disable_cache: false
  main: |
    from returns.pointfree import unify
    from returns.context import ReaderFutureResult

    def test(arg: int) -> ReaderFutureResult[int, bool, float]:
        ...

    x: ReaderFutureResult[int, str, float]
    reveal_type(unify(test)(x))  # N: Revealed type is 'returns.context.requires_context_future_result.RequiresContextFutureResult[builtins.int, Union[builtins.str*, builtins.bool], builtins.float]'
```

Reply:
You can just create a new issue out of it for now.

Reply:
https://github.com/dry-python/returns/issues/637